packages/legend-application-query/src/components/QueryBuilderResultPanel.tsx

```diff
@@ -474,23 +489,18 @@ export const QueryBuilderResultPanel = observer(
             )}
             <button
               className="query-builder__result__execute-btn"
-              onClick={execute}
-              disabled={
-                resultState.isExecutingQuery || resultState.isGeneratingPlan
-              }
+              onClick={isRunningQuery ? cancelQuery : runQuery}
               tabIndex={-1}
             >
               <div className="query-builder__result__execute-btn__label">
                 <PlayIcon className="query-builder__result__execute-btn__label__icon" />
                 <div className="query-builder__result__execute-btn__label__title">
-                  Run Query
+                  {isRunningQuery ? 'Cancel' : 'Run Query'}
```

Comment:
--> `Stop` and also we might want to change the styling here, certain operation in the dropdown wouldn't make sense anymore`. As such, I think we could consider replacing this button by another one while executing. What do you think @MauricioUyaguari @YannanGao-gs?